### ISSUE TITLE
Train on your own 512 x 512 size image

### ISSUE THREAD

**Issue by zts12** [htcr/sam_road#32]
For 512*512 images, do I need to modify the config settings for training on my own dataset? (This is the SpaceNet configuration). Also, do you have any advice on the training batch, the results of 30 sessions are not ideal? Thank you very much for your work on the SAM-Road project, and look forward to your answers in your busy schedule, thank you.

**Comment by htcr** [comment]
I think the CityScale setup defaults to 512x512 patch size. Can you try that? Batch size depends on your GPU memory, I think you can start with the largest batch size you can get away with - then tune LR properly to make sure it converges. It may need some trail-and-error.

**Comment by zts12** [comment]
Thank you very much for your answer, this is the result of my current road extraction with my own 0.5 meter resolution image using the configuration of SpaceNet, not very ideal, do you have any suggestions?
![Uploading bj000035_mask_road_mask.png…]()

**Comment by zts12** [comment]
![bj000036_mask_road_mask](https://github.com/user-attachments/assets/96a6782b-9e13-4ab1-ac55-7eef7b37950d)
![bj000041_mask_road_mask](https://github.com/user-attachments/assets/b0ea54cd-795c-4706-b704-b1aa8e261deb)
![bj000046_mask_road_mask](https://github.com/user-attachments/assets/d76e4276-f768-4a9a-8aba-b0dbd9a4e894)
![bj000049_mask_road_mask](https://github.com/user-attachments/assets/01530fa4-9694-4396-91e6-1cf5f2f88ae1)
![Uploading bj000056_mask_road_mask.png…]()

**Comment by htcr** [comment]
I think our released model takes 1.0m/pixel images. Can you try resizing your images to that resolution? Also, have you fine-tuned on your own dataset? How large was your dataset?

**Comment by htcr** [comment]
Also did you correctly load the pre-trained SAM ckpts?

**Comment by zts12** [comment]
Thank you again for your suggestion, I used the 0.5 meter resolution image to crop it to 512*512 image size, and also loaded the ckpt of the pre-trained SAM, and also adjusted the learning rate, and re-trained, but the effect is not justified by the good effect of the two datasets in the original paper, the dataset has a total of 3065 pieces, 2453 for training, 459 for testing, 153 for verification, according to the scale of Spacenet for the data division Each image is 512 and contains the corresponding required graph data. So if I change the image to 1 meter resolution, will the end result be improved? The results of the test are like this, and it is not clear whether the thresholds of key points and roads should also be modified?
======= Finding best thresholds ======
======= keypoint ======
======= Finding best thresholds ======
======= keypoint ======
Best threshold 0.01090240478515625, P=0.0 R=0.0 F1=nan
======= road ======
Best threshold 0.01090240478515625, P=0.0 R=0.0 F1=nan
======= road ======
Best threshold 0.0965576171875, P=0.0 R=0.0 F1=nan
======= topo ======
Best threshold 0.0965576171875, P=0.0 R=0.0 F1=nan

**Comment by htcr** [comment]
Hi, I think if you are fine-tuning from the original SAM ckpt (not the ones I released), resolution is less crucial. How does the images look like in general? The numbers you shown seems to suggest the model did not converge at all. The size of the dataset sounds reasonable, can you try the following options:

1) Debug the label generation logic. Does the GT masks look reasonable?
2) See if the model can just overfit one example. If not, maybe some hyperparameters are wrong.
3) Try different batch sizes / learning rates. 
4) Apply some data augmentation. In SAM-Road paper, random cropping and rotation were applied.
5) Try to zero some loss terms to find which one is exploding.

Good luck with your experiments!

**Comment by zts12** [comment]
Sorry for the late reply, thank you for your suggestion, I will follow your suggestion to carry out the experiment, I am a graduate student in a university, and the current direction of study is to use high-resolution remote sensing images for road extraction, thank you for communicating with you, can you add WeChat, my WeChat account is 18837621961, I will be honored.

**Comment by EchoQiHeng** [comment]
I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.

**Comment by zts12** [comment]
Thank you for your work sharing, but I also used DeepGlobe for training and testing, I cropped it to a 512*512 image, and trained and tested, but the result is not very ideal, but the clear road that can be extracted is incomplete The effect is not very good, can I consult your config settings and the division rules of the dataset? Or is there some other modification work and configuration work that I haven't noticed? Thanks for your answer.

> I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.

**Comment by EchoQiHeng** [comment]
> Thank you for your work sharing, but I also used DeepGlobe for training and testing, I cropped it to a 512*512 image, and trained and tested, but the result is not very ideal, but the clear road that can be extracted is incomplete The effect is not very good, can I consult your config settings and the division rules of the dataset? Or is there some other modification work and configuration work that I haven't noticed? Thanks for your answer.
> 
> > I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.

I have demonstrated the visualization results on the DeepGlobe validation set, and I believe the model has converged and is functioning as expected. It seems I did not make any specific configuration settings for DeepGlobe. Of course, modifications to the **SatMapDataset** were necessary, and my process primarily involved cropping and augmentation. Please carefully check your RGB images and the corresponding GT Mask.
![iou](https://github.com/user-attachments/assets/72ee9e2a-d0a8-45f2-b652-06ad19428f53)
![pred](https://github.com/user-attachments/assets/0cd1ad92-b764-4cc0-a6ad-8c147d0fe37f)
Additionally, I have displayed the IoU during the training process. Please provide more details and results from your experiments to facilitate further debugging.
![rgb](https://github.com/user-attachments/assets/c9e1b8de-78cd-4983-abac-4a371f779979)

**Comment by immarshmellow** [comment]
> > 感谢大家的分享，不过我也用 DeepGlobe 进行了训练和测试，我裁剪成了一张 512*512 的图片，进行了训练和测试，但是结果不是很理想，但是可以提取的清路效果不是很好，可以查阅一下你的配置设置和数据集的划分规则吗？或者还有其他一些我没有注意到的修改工作和配置工作？谢谢你的回答。
> > > 我在 DeepGlobe 数据集上训练了 SAM，结果令人信服，因此我相信 SAM 是稳健的。请仔细检查您的代码。
> 
> 我已经在 DeepGlobe 验证集上演示了可视化结果，我相信该模型已经收敛并按预期运行。我似乎没有为 DeepGlobe 进行任何特定的配置设置。当然，对 **SatMapDataset** 的修改是必要的，我的过程主要涉及裁剪和增强。请仔细检查您的 RGB 图像和相应的 GT 蒙版。 ![iou](https://private-user-images.githubusercontent.com/96250537/374777858-72ee9e2a-d0a8-45f2-b652-06ad19428f53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MDAzMDcsIm5iZiI6MTczMzkwMDAwNywicGF0aCI6Ii85NjI1MDUzNy8zNzQ3Nzc4NTgtNzJlZTllMmEtZDBhOC00NWYyLWI2NTItMDZhZDE5NDI4ZjUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDA2NTMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4ODA3YTM2ZWM0NDI3ZjI0ODQ2NDJmNWY5NWM0NDBjYWJmMDk0NjA3MDAxNTU2NGMxMjhlMmM2YjMyMDc4N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.I3xD3CYbImr4RJPZ19E3orWtI_KZG4_KVLS_-QrPSTU) ![pred](https://private-user-images.githubusercontent.com/96250537/374777863-0cd1ad92-b764-4cc0-a6ad-8c147d0fe37f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MDAzMDcsIm5iZiI6MTczMzkwMDAwNywicGF0aCI6Ii85NjI1MDUzNy8zNzQ3Nzc4NjMtMGNkMWFkOTItYjc2NC00Y2MwLWE2YWQtOGMxNDdkMGZlMzdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDA2NTMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2NDMxZGYxMGZhYzgyZWZjYTRhNTBjMzVlMDkwNGFhNTgyZjI4MzhhNzkwYWU5OWQ4MTcxMTk0NTAzNmM0MWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NEEyu-5l9ccr9A5MWsgcrjm14TZxmplmk37FA_o3NdQ)此外，我在训练过程中还展示了 IoU。请提供实验的更多详细信息和结果，以便于进一步调试。 ![rgb](https://private-user-images.githubusercontent.com/96250537/374777867-c9e1b8de-78cd-4983-abac-4a371f779979.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5MDAzMDcsIm5iZiI6MTczMzkwMDAwNywicGF0aCI6Ii85NjI1MDUzNy8zNzQ3Nzc4NjctYzllMWI4ZGUtNzhjZC00OTgzLWFiYWMtNGEzNzFmNzc5OTc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjExVDA2NTMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiOTViOTRiZjIwYWJlYzUxN2VkNDI5NmVmZmZmNTQ5NWRiNWVmNmNhNjVkZGU2NDFlZmZjYTAwY2RkYmVkMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aEzGEYP_eJL7152q7EWtQyj97dOgGLRRJYujDfEjPZE)

Hi, I would like to ask about the exact steps for training and testing with DeepGlobe. I don't understand much of this stuff and appreciate any help that can be provided.

**Comment by tan-maker** [comment]
> > Thank you for your work sharing, but I also used DeepGlobe for training and testing, I cropped it to a 512*512 image, and trained and tested, but the result is not very ideal, but the clear road that can be extracted is incomplete The effect is not very good, can I consult your config settings and the division rules of the dataset? Or is there some other modification work and configuration work that I haven't noticed? Thanks for your answer.
> > > I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.
> 
> I have demonstrated the visualization results on the DeepGlobe validation set, and I believe the model has converged and is functioning as expected. It seems I did not make any specific configuration settings for DeepGlobe. Of course, modifications to the **SatMapDataset** were necessary, and my process primarily involved cropping and augmentation. Please carefully check your RGB images and the corresponding GT Mask. ![iou](https://github.com/user-attachments/assets/72ee9e2a-d0a8-45f2-b652-06ad19428f53) ![pred](https://github.com/user-attachments/assets/0cd1ad92-b764-4cc0-a6ad-8c147d0fe37f) Additionally, I have displayed the IoU during the training process. Please provide more details and results from your experiments to facilitate further debugging. ![rgb](https://github.com/user-attachments/assets/c9e1b8de-78cd-4983-abac-4a371f779979)

Hello, since DeepGlobe does not provide the region_{}_refine_gt_graph.p files, how did you generate them?

**Comment by tan-maker** [comment]
> I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.

deepglobe dataset has no graph only mask

**Comment by xiaofei-guan** [comment]
> > Thank you for your work sharing, but I also used DeepGlobe for training and testing, I cropped it to a 512*512 image, and trained and tested, but the result is not very ideal, but the clear road that can be extracted is incomplete The effect is not very good, can I consult your config settings and the division rules of the dataset? Or is there some other modification work and configuration work that I haven't noticed? Thanks for your answer.
> > > I trained SAM on the DeepGlobe dataset and the results were convincing, so I believe SAM is robust. Please carefully check your code.
> 
> I have demonstrated the visualization results on the DeepGlobe validation set, and I believe the model has converged and is functioning as expected. It seems I did not make any specific configuration settings for DeepGlobe. Of course, modifications to the **SatMapDataset** were necessary, and my process primarily involved cropping and augmentation. Please carefully check your RGB images and the corresponding GT Mask. ![iou](https://github.com/user-attachments/assets/72ee9e2a-d0a8-45f2-b652-06ad19428f53) ![pred](https://github.com/user-attachments/assets/0cd1ad92-b764-4cc0-a6ad-8c147d0fe37f) Additionally, I have displayed the IoU during the training process. Please provide more details and results from your experiments to facilitate further debugging. ![rgb](https://github.com/user-attachments/assets/c9e1b8de-78cd-4983-abac-4a371f779979)

Hi, there are no graph labels in deepglobe dataset, from the demenstraion above you provide, it's a road mask not graph, and the curve picture is also road_iou relation, so I guess the model you trained was actually segement anything model instead of sam_road model, or you have labeled the data by yourselves for training sam_road?
